### ISSUE TITLE
Support override and private typescript modifiers

### ISSUE THREAD

**Issue by Lusito** [gajus/eslint-plugin-jsdoc#759]
## Motivation

TypeScript supports override and private keywords, but overrideReplacesDocs and ignorePrivate do not work with them. They require the jsdoc annotations.

## Current behavior

```typescript

export abstract class A {
    /**
     * The update method called every tick.
     *
     * @param deltaTime The time passed since last frame in seconds.
     */
    public abstract update(deltaTime: number): void;

   // Missing JSDoc comment. eslint (jsdoc/require-jsdoc)
    private doStuff() {
        // ....
    }
}

export class B extends A {
   // Missing JSDoc comment. eslint (jsdoc/require-jsdoc)
    public override update(deltaTime: number) {
        // ...
    }
}
```

## Desired behavior

```typescript

export abstract class A {
    /**
     * The update method called every tick.
     *
     * @param deltaTime The time passed since last frame in seconds.
     */
    public abstract update(deltaTime: number): void;

   // OK
    private doStuff() {
        // ....
    }
}

export class B extends A {
   // OK
    public override update(deltaTime: number) {
        // ...
    }
}
```

## Alternatives considered

- Manually adding `@private` and `@override` to the jsdoc.
- Disabling the eslint rule on a per line basis

Thanks for the good work!

**Comment by brettz9** [comment]
You can use the option `contexts` to provide AST which only finds non-private methods; but do not enable the plain `MethodDefinition` option (or `FunctionExpression` option):

`contexts: ['MethodDefinition:not([accessibility="private"] > FunctionExpression']`

**Comment by brettz9** [comment]
Closing as that should resolve, but feel free to comment further if needed.

**Comment by Lusito** [comment]
Thanks @brettz9, I didn't know about contexts. But I can't get it to work:

```javascript
        "jsdoc/require-jsdoc": [
            "warn",
            {
                publicOnly: {
                    ancestorsOnly: true,
                    esm: true,
                    cjs: true,
                    window: false,
                },
                require: {
                    ArrowFunctionExpression: true,
                    ClassDeclaration: true,
                    ClassExpression: true,
                    FunctionDeclaration: true,
                    FunctionExpression: true,
                    MethodDefinition: true,
                },
                contexts: [
                    'MethodDefinition:not([accessibility="private"] > FunctionExpression)',
                    'MethodDefinition:not([override=true] > FunctionExpression)'
                ],
            },
        ],
```

It still complains.

**Comment by Lusito** [comment]
Ah, I needed to disable the MethodDefinition, set it manually with overrides and also combine the two nots into one:

```javascript
        "jsdoc/require-jsdoc": [
            "warn",
            {
                publicOnly: {
                    ancestorsOnly: true,
                    esm: true,
                    cjs: true,
                    window: false,
                },
                require: {
                    ArrowFunctionExpression: true,
                    ClassDeclaration: true,
                    ClassExpression: true,
                    FunctionDeclaration: true,
                    FunctionExpression: true,
                    MethodDefinition: false,
                },
                contexts: ['MethodDefinition:not([accessibility="private"]):not([override=true]) > FunctionExpression'],
            },
        ],
```
